mlrun/data_types/to_pandas.py

```diff
@@ -145,18 +137,7 @@ def toPandas(spark_df):
                             )
                     return pdf
                 else:
-                    corrected_panda_types = {}
-                    for index, field in enumerate(spark_df.schema):
-                        pandas_type = _to_corrected_pandas_type(field.dataType)
-                        corrected_panda_types[tmp_column_names[index]] = (
-                            np.object0 if pandas_type is None else pandas_type
-                        )
-
-                    pdf = pd.DataFrame(columns=tmp_column_names).astype(
-                        dtype=corrected_panda_types
-                    )
-                    pdf.columns = spark_df.columns
-                    return pdf
+                    return pd.DataFrame.from_records([], columns=spark_df.columns)
```

Comment:
Regardless of whether the DataFrame is empty or not, ensure that the column types are preserved.